### PR TITLE
Generate single-page HTML, PDF, manpage from documentation

### LINKED ISSUE #3670: Generate PDF again from documention (with Sphinx)
There was a request to have PDF documentation for printing. I'll try to get to this soon.

(I've also closed the previous PDF issue: #332.)

### PR DESCRIPTION
**Pull request by msimberg**
Fixes #3670.

### CODE REVIEW

File: cmake/templates/conf.py.in [hunk 1]
```diff
@@ -595,16 +595,26 @@ htmlhelp_basename = 'HPXdoc'
...
+}
 
 # Grouping the document tree into LaTeX files. List of tuples
 # (source start file, target name, title,
 #  author, documentclass [howto, manual, or own class]).
 latex_documents = [
     (master_doc, 'HPX.tex', u'HPX Documentation',
-     u'The Ste\\textbar{}\\textbar{}ar Group', 'manual'),
+     u'The STE\\textbar{}\\textbar{}AR Group', 'manual'),
 ]
```
Comment (hkaiser) [inline cmake/templates/conf.py.in]: FWIW, we've used `{STE\raisebox{.05em}{\hspace{.1em}$|$\hspace{0.1em}$|$\hspace{.02em}}AR}` before to typeset `STE||AR`

Comment (msimberg) [reply cmake/templates/conf.py.in]: That's awfully specific, but I can change that ;) Thanks!

Comment (hkaiser) [reply cmake/templates/conf.py.in]: There is nothing more important than names and logos! ;-)

**Comment by msimberg** [comment]
Looks like this would be ready to go now. PDF docs: https://stellar-group.github.io/hpx/docs/sphinx/branches/pdf-docs/pdf/HPX.pdf. Single-page HTML docs: https://stellar-group.github.io/hpx/docs/sphinx/branches/pdf-docs/singlehtml/index.html. 

I'll add links to the PDF and single-page version to the downloads pages once they get built on master.

**Comment by alexmyczko** [comment]
just a note on manpage generation. it get's installed in `/usr/share/man/` but really should go into `/usr/share/man/man1/` and be `gzip -9`'d

otherwise man will not find it. see man man, or /etc/manpath.config

and then with the 1.3.0 release generated manpage i get:
`W: libhpx-dev: manpage-has-errors-from-man usr/share/man/man1/hpx.1.gz 3761: warning [p 41, 0.8i]: cannot adjust line`

**Comment by msimberg** [comment]
Thanks @alexmyczko. The path we can definitely change. I assume you gzipped the manpage to arrive at `hpx.1.gz`? The gzipping should also be doable. Would you have an idea of what that warning is trying to tell us?

**Comment by alexmyczko** [comment]
@msimberg No clue :) but this command 

```
LC_ALL=en_US.UTF-8 MANROFFSEQ='' MANWIDTH=80 \
            man --warnings -E UTF-8 -l -Tutf8 -Z /usr/share/man/man1/hpx.1.gz >/dev/null
```

shows a lot more of them, command taken from:

https://lintian.debian.org/tags/manpage-has-errors-from-man.html

**Pull request merged**